error[E0005]: refutable pattern in local binding: `Err(_)` not covered
  --> $DIR/feature-gate-exhaustive-patterns.rs:8:9
   |
LL |     let Ok(_x) = foo();
   |         ^^^^^^ pattern `Err(_)` not covered
   |
   = note: `let` bindings require an "irrefutable pattern", like a `struct` or an `enum` with only one variant
   = note: for more information, visit https://doc.rust-lang.org/book/ch18-02-refutability.html
note: `Result<u32, !>` defined here
  --> $SRC_DIR/core/src/result.rs:LL:COL
   |
LL | / pub enum Result<T, E> {
LL | |     /// Contains the success value
LL | |     #[lang = "Ok"]
LL | |     #[stable(feature = "rust1", since = "1.0.0")]
...  |
LL | |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
   | |     ^^^ not covered
LL | | }
   | |_-
   = note: the matched value is of type `Result<u32, !>`
help: you might want to use `if let` to ignore the variant that isn't matched
   |
LL |     let _x = if let Ok(_x) = foo() { _x } else { todo!() };
   |     +++++++++++                    +++++++++++++++++++++++
help: alternatively, on nightly, you might want to use `#![feature(let_else)]` to handle the variant that isn't matched
   |
LL |     let Ok(_x) = foo() else { todo!() };
   |                        ++++++++++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0005`.
